### PR TITLE
MATLAB integration №2

### PR DESCRIPTION
**Pull request by RawrLion**

### CODE REVIEW

File: integration/matlab/README.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+![](resources/prefdir.png)
+- Close MatLab if it is running.
+- Navigate to the folder (output of `prefdir` command) and create a file called `javaclasspath.txt` in the folder.
+- Open javaclasspath.txt. Add the full path to the database driver JAR file in `javaclasspath.txt`. The full path includes the path to the folder where you downloaded the JAR file from the database provider and the JAR file name. For example, `/home/alex/Downloads/atsd-jdbc-1.2.22-SNAPSHOT-DEPS.jar`
+- Save and close `javaclasspath.txt`.
+- Restart MatLab.
+
+### Dynamic:
+- Run the javaaddpath(dpath) command in the Command Window, where dpath is a path to database driver JAR file.
```
Comment (rodionos) [inline integration/matlab/README.md]: Change to the latest driver version, e.g. 1.2.22

File: integration/matlab/README.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+- [Load Data](#load-data)
+- [Write Data](#write-data)
+- [Calculate Derived Series](#calculate-derived-series)
+
+## Install ATSD driver
+
+There's two ways of connecting ATSD JDBC driver to MatLab: static and dynamic
+
+### Static:
+- Run the `prefdir` command in the Command Window. The output of this command is a file path to a folder on your computer.
```
Comment (rodionos) [inline integration/matlab/README.md]: Clarify if this is MatLab or OS Command Window?

File: integration/matlab/README.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+![](resources/prefdir.png)
+- Close MatLab if it is running.
+- Navigate to the folder (output of `prefdir` command) and create a file called `javaclasspath.txt` in the folder.
+- Open javaclasspath.txt. Add the full path to the database driver JAR file in `javaclasspath.txt`. The full path includes the path to the folder where you downloaded the JAR file from the database provider and the JAR file name. For example, `/home/alex/Downloads/atsd-jdbc-1.2.22-SNAPSHOT-DEPS.jar`
```
Comment (rodionos) [inline integration/matlab/README.md]: Downloads folder may get cleaned up. I think it's better to copy the driver to a permanent folder somewhere in matlab installation directory.

File: integration/matlab/README.md (hunk 1 above)
Comment (rodionos) [inline integration/matlab/README.md]: highlight `dpath` as a variable, and not regular text

File: integration/matlab/README.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+- Save and close `javaclasspath.txt`.
+- Restart MatLab.
+
+### Dynamic:
+- Run the javaaddpath(dpath) command in the Command Window, where dpath is a path to database driver JAR file.
+
+Example:
+
+![](resources/java_add_path.png)
```
Comment (rodionos) [inline integration/matlab/README.md]: The Dynamic option seems easier. Shall we just keep Dynamic for simplicity? What is the difference between these options?

File: integration/matlab/README.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+
+## Configure ATSD Connection
+
+### ATSD connection via Database Explorer
+
+- Click New - JDBC.
+- Vendor - OTHER.
+- Driver is a class path of ATSD JDBC driver (com.axibase.tsd.driver.jdbc.AtsdDriver).
+- URL is a JDBC URL like ```jdbc:axibase:atsd:https://ATSD_HOSTNAME:8443/api/sql;catalog=atsd;tables="WANTED_METRIC";expandTags=true;trustServerCertificate=true```
```
Comment (rodionos) [inline integration/matlab/README.md]: WANTED_METRIC?

Comment (rodionos) [inline integration/matlab/README.md]: Add link to atsd jdbc docs where parameters are explained

File: integration/matlab/README.md [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+- [Configure ATSD Connection](#configure-atsd-connection)
+- [Verify Connection](#verify-connection)
+- [Review Tables in Database Explorer](#review-tables-in-database-explorer)
+- [Load Data](#load-data)
+- [Write Data](#write-data)
+- [Calculate Derived Series](#calculate-derived-series)
+
+## Install ATSD driver
+
+There's two ways of connecting ATSD JDBC driver to MatLab: static and dynamic
```
Comment (rodionos) [inline integration/matlab/README.md]: Specify where the driver can be downloaded from

File: integration/matlab/README.md [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+- Leave Username and Password fields empty -- there are only for test connection.
+- Now connect to ATSD using "Data source name" and login with password to open Database Browser window.
+
+Example:
+
+![](resources/new_jdbc_data_source.png)
+
+ATSD_HOSTNAME is a hostname address of ATSD instance you want connect to
+
+TABLE_NAME_FILTER is a metric template (metrics that meet pattern requirement will be displayed in Database Browser)
```
Comment (rodionos) [inline integration/matlab/README.md]: TABLE_NAME_FILTER is a list of comma-separated metrics or metric expressions to be displayed as tables in the MatLab Database Browser.

File: integration/matlab/README.md [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+
+TABLE_NAME_FILTER examples:
+- `*java*` for metrics that contains word `java`
+- `custom.metric*` for metrics which name starts with `custom.metric`
+- `*2017` for metrics which name ends with `2017`
+
+URL for http connection example:  
+```jdbc:axibase:atsd:http://ATSD_HOSTNAME:8088/api/sql;catalog=atsd;tables="TABLE_NAME_FILTER";expandTags=true```
```
Comment (rodionos) [inline integration/matlab/README.md]: single quotes are sufficient. remove 'catalog' parameter from everywhere, it's not necessary

File: integration/matlab/README.md [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+![](resources/data_example.png)
+
+Example of inserting data into ATSD (assuming `conn_atsd` is set already and `data` variable has needed values):
+
+```matlab
+colnames = {'datetime', 'entity', 'value'};
+insert(conn_atsd, 'METRIC_NAME', colnames, data);
+```
+
+`colnames` is a cell array which describes names and order of columns in payload
```
Comment (rodionos) [inline integration/matlab/README.md]: Add a note explaining what is METRIC_NAME

File: integration/matlab/README.md (hunk 8 above)
Comment (RawrLion) [inline integration/matlab/README.md]: Absent of 'catalog' parameter leads to NullPointerException.
Stacktrace:
```
java.lang.NullPointerException
	at com.mathworks.toolbox.database.gui.DataAccess.getContents(DataAccess.java:423)
	at com.mathworks.toolbox.database.gui.DataAccess.setTreeData(DataAccess.java:226)
	at com.mathworks.toolbox.database.gui.DataAccess$3$1.runOnAWTEventDispatchThread(DataAccess.java:832)
	at com.mathworks.jmi.MatlabWorker$1.run(MatlabWorker.java:67)
	at java.awt.event.InvocationEvent.dispatch(Unknown Source)
	at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.awt.EventQueue.access$200(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(Unknown Source)
	at java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.awt.WaitDispatchSupport$2.run(Unknown Source)
	at java.awt.WaitDispatchSupport$4.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.awt.WaitDispatchSupport.enter(Unknown Source)
	at java.awt.Dialog.show(Unknown Source)
	at com.mathworks.mwswing.MJDialog.show(MJDialog.java:311)
	at com.mathworks.toolbox.database.gui.DataAccess.createLoginDialog(DataAccess.java:905)
	at com.mathworks.toolbox.database.gui.DataAccess.access$000(DataAccess.java:33)
	at com.mathworks.toolbox.database.gui.DataAccess$1.runOnAWTEventDispatchThread(DataAccess.java:601)
	at com.mathworks.jmi.MatlabWorker$1.run(MatlabWorker.java:67)
	at java.awt.event.InvocationEvent.dispatch(Unknown Source)
	at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.awt.EventQueue.access$200(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(Unknown Source)
	at java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.run(Unknown Source)
```

File: integration/matlab/README.md (hunk 4 above)
Comment (RawrLion) [inline integration/matlab/README.md]: Dynamic class path must be used on every startup of MatLab. It's a feature for developers mostly

**Pull request merged**